grabbit/src/main/groovy/com/twcable/grabbit/client/batch/steps/validation/ValidJobDecider.groovy

```diff
@@ -0,0 +1,83 @@
+package com.twcable.grabbit.client.batch.steps.validation
+
+import com.twcable.grabbit.client.batch.ClientBatchJob
+import com.twcable.grabbit.jcr.JcrUtil
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import org.apache.sling.jcr.api.SlingRepository
+import org.springframework.batch.core.ExitStatus
+import org.springframework.batch.core.JobExecution
+import org.springframework.batch.core.StepExecution
+import org.springframework.batch.core.job.flow.FlowExecutionStatus
+import org.springframework.batch.core.job.flow.JobExecutionDecider
+
+import javax.annotation.Nonnull
+import javax.jcr.PathNotFoundException
+import javax.jcr.RepositoryException
+import javax.jcr.Session
+
+/*
+ * Copyright 2015 Time Warner Cable, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@CompileStatic
+@Slf4j
+class ValidJobDecider implements JobExecutionDecider {
+
+    private SlingRepository slingRepository
+
+    final static FlowExecutionStatus JOB_VALID = new FlowExecutionStatus("VALID")
+    final static FlowExecutionStatus JOB_INVALID = new FlowExecutionStatus("INVALID")
+
+
+    ValidJobDecider(@Nonnull final SlingRepository slingRepository) {
+        this.slingRepository = slingRepository
+    }
+
+    /**
+     * Determines if the job to be executed is a valid job
+     *
+     * For example, are we being asked to sync a path for which a path has no existing parents?
+     * If so, we should label this scenario as an invalid job; and not attempt to write "dirty data"
+     *
+     * @param jobExecution a job execution
+     * @param stepExecution the latest step execution (may be null)
+     * @return the exit status code.  Is it a valid or invalid job?
+     */
+    @Override
+    FlowExecutionStatus decide(JobExecution jobExecution, StepExecution stepExecution) {
+        final String jobPath = jobExecution.jobParameters.getString(ClientBatchJob.PATH)
+        //For processing, remove trailing /
+        jobPath = jobPath.replaceFirst(/\/$/, '')
+        //Get the parent's path (if applicable) and determine if it exists already
+        final parts = jobPath.split('/')
+        //No parent, so nothing to worry about
+        if(parts.length <= 2) return JOB_VALID
+
+        final parentPath = jobPath - "/${parts[-1]}"
+        final Session session = JcrUtil.getSession(slingRepository, 'admin')
```

Comment:
1. You think we should use `clientUsername` from the Job Parameters instead of `admin`. Cannot really pin point a use case, but we have now removed all usages of `admin` user.
2. Should close out the `session` when done with it.

P.S. : You get away with both of this if you use the `session` in the `ThreadLocal`.